_includes/lang.html

```diff
@@ -1,7 +1,9 @@
 {% comment %}
   Detect appearance language and return it through variable "lang"
 {% endcomment %}
-{% if site.data.locales[site.lang] %}
+{% if site.data.locales[page.lang] %}
+  {% assign lang = page.lang %}
+{% else if site.data.locales[site.lang] %}
   {% assign lang = site.lang %}
 {% else %}
   {% assign lang = 'en' %}
```

Comment:
```diff
-{% if site.data.locales[site.lang] %}
-  {% assign lang = site.lang %}
-{% else %}
+{% assign lang = page.lang | default: site.lang %}
+
+{% unless site.data.locales[lang] %}
   {% assign lang = 'en' %}
-{% endif %}
+{% endunless %}
```

Reply:
The `lang.html` can be optimized.

Reply:
This logic is slightly different - if the page lang is invalid then my way would use `site.lang`, whereas this way will use `en`. But either argument is valid, so simpler is better!

Reply:
> This logic is slightly different - ...

You're right, so please ignore my advice and maintain the changes you submitted the first time.

```diff
-{% if site.data.locales[site.lang] %}
+{% if site.data.locales[page.lang] %}
+  {% assign lang = page.lang %}
+{% else if site.data.locales[site.lang] %}
   {% assign lang = site.lang %}
 {% else %}
   {% assign lang = 'en' %}
```